src/frauddetectionservice/src/main/kotlin/frauddetectionservice/main.kt

```diff
@@ -46,8 +59,30 @@ fun main() {
                     val newCount = accumulator + 1
                     val orders = OrderResult.parseFrom(record.value())
                     logger.info("Consumed record with orderId: ${orders.orderId}, and updated total count to: $newCount")
+                    if (getFeatureFlagValue("kafkaQueueProblems") > 0) {
```

Comment:
My Kotlin isn't that great, but this seems to be checking the feature flag after the record is consumed from Kafka. I think we want it before so internal Kafka and message metrics indicate a lag in consuming the message.

Reply:
> My Kotlin isn't that great, but this seems to be checking the feature flag after the record is consumed from Kafka. I think we want it before so internal Kafka and message metrics indicate a lag in consuming the message.

You are correct that it is sleeping after the record is consumed (i can change this if you would like to have the sleep before the claim). However, in practice we are still producing a lag since the consumer is busy with the sleep after having consumed the message. This results in the consumer only being able to claim a record approx. every second.

Reply:
I would still do this before. There are span attributes we automatically attach for show consumer lag time which I would hope will show an increase when this flag is enabled.

Reply:
Alright, will move the sleep up!